luigi/contrib/spark.py

```diff
@@ -237,8 +237,6 @@ class PySparkTask(SparkSubmitTask):
 
     # Path to the pyspark program passed to spark-submit
     app = os.path.join(os.path.dirname(__file__), 'pyspark_runner.py')
-    # Python only supports the client deploy mode, force it
-    deploy_mode = "client"
```

Comment:
`deploy_mode = "client"` overwrites `deploy_mode` in `SparkSubmitTask`. It was here to force deploy mode to be `client` since previously cluster deploy mode wasn't supported. Now that we do support it, there is no need to pin it to `client` only.